pints/_log_pdfs.py

```diff
@@ -115,6 +115,178 @@ def sample(self, n=1):
         raise NotImplementedError
 
 
+class PooledLogPDF(LogPDF):
+    r"""
+    Creates an object of :class:LogPDF, with a log-pdf which is the sum of the
+    individual log-pdfs, while pooling selected parameters.
+
+    This :class:`LogPDF` requires that all :class:`LogPDF` objects have the
+    same number of parameters.
+
+    This is useful for e.g. modelling the time-series of multiple individuals
```

Comment:
Either the first sentence needs to be clearer, or there needs to be more explanation here (before going into the use for this class).

I think we might have to use numbers/equations in the initial explanation, just to get things clear. Because I can't think of standard terminology that will let us explain this in words?
E.g. something like 

```
    Combines ``m`` :class:`LogPDFs<pints.LogPDF>`, each with ``n`` parameters,
    into a single ``LogPDF`` where one or more parameters are "pooled" (i.e.
    have the same value for each ``LogPDF``), so that the resulting combined 
    ``LogPDF`` has ``m * n_unpooled + n_pooled`` parameters.
```

Or maybe "where k parameters are pooled" so you can write ``m * (n - k) + k`` ? 

(note 1: You can do :class:`Hello<RealClassName>` to make the link say hello)

(note 2: No need to do a :class:`X` twice in the same sentence, just use ``X`` for the second one)

Reply:
That's much clearer!